CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor/Contribution.php

```diff
@@ -35,10 +35,22 @@ public function getRecurringContribution() {
    * depending on the instalments count.
    */
   public function createPaymentPlan() {
+    $this->alterRecievedDate();
     $this->createRecurringContribution();
     $this->alterFirstContributionParameters();
   }
 
+  /**
+   * Set Contribution receive date as Membership start date.
+   */
+  private function alterRecievedDate() {
```

Comment:
this does not seem that it will work well when we renew with "don't update membership start date on renewal" setting